extractor/src/main/java/org/schabi/newpipe/extractor/services/media_ccc/extractors/MediaCCCStreamExtractor.java

```diff
@@ -167,18 +162,119 @@ public List<VideoStream> getVideoStreams() throws ExtractionException {
                 // information to decide whether two streams are similar. Hence that method would
                 // always return false, e.g. even for different language variations.
                 videoStreams.add(new VideoStream.Builder()
-                        .setId(recording.getString("filename", ID_UNKNOWN))
-                        .setContent(recording.getString("recording_url"), true)
+                        .setId(recording.filename)
+                        .setContent(recording.url, true)
                         .setIsVideoOnly(false)
                         .setMediaFormat(mediaFormat)
-                        .setResolution(recording.getInt("height") + "p")
+                        .setResolution(recording.height + "p")
                         .build());
-            }
         }
 
         return videoStreams;
     }
 
+    public List<MediaCCCRecording> getRecordings() {
```

Comment:
Sorry for the late review. You should probably cache the value returned by `getRecordings()` to avoid recalculating it in `getVideoStreams()`, `getAudioStreams()` and `getVideoOnlyStreams()`

Reply:
Oh, but it’s not doing a network request here, the `data` object will be used for both function calls as long as the object is preserved between them. The Json extracting code itself should be on the order of microseconds.

Reply:
True, it's not important, it just feels bad to do the same extraction 3 times